index.js

```diff
@@ -295,7 +295,11 @@ Shopify.prototype.graphql = function graphql(data, variables) {
 
             if (res.body.errors) {
```

Comment:
I think it's better to add the `Array.isArray()` check here, in the `if` condition. This branch should only cover the 2xx case. The same check should also be done in the `if` condition on line 323.

Reply:
Sure thing, updated to check the response body and applied same check to `decorateError`